iceoryx_posh/include/iceoryx_posh/popo/smart_chunk.hpp

```diff
@@ -174,6 +173,10 @@ class SmartChunk
     ///
     const mepoo::ChunkHeader* getChunkHeader() const noexcept;
 
+    /// @note used by the producer to release the chunk ownership from the `SmartChunk` after publishing the chunk and
+    /// therefore preventing the invocation of the custom deleter
+    T* release() noexcept;
```

Comment:
Was there a special reason why this is made public? With `release` one takes the ownership of the chunk but with the typed API there is no way to free the sample so this is a dangerous operation which might lead to a chunk leak